### PR TITLE
Improved branch wpmlcore-6607 with attribute `search-method`

### PR DESCRIPTION
**Pull request by strategio**
See https://github.com/OnTheGoSystems/wpml-page-builders-gutenberg/pull/40#pullrequestreview-239135373.

https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6607

### CODE REVIEW

**Comment by andreasciamanna** [comment]
@strategio looks good to me if I understand correctly how you implemented it.

Updating the XML validation tests would help better understanding: will you do that?

**Review by andreasciamanna — APPROVED** [review]

**Comment by strategio** [comment]
@OnTheGoSystems/wpml one more review please.

**Review by andreasciamanna — APPROVED** [review]

**Comment by strategio** [comment]
@OnTheGoSystems/wpml one more review please!

**Comment by strategio** [comment]
@brucepearson, if you are still awake, can you give your review here (this is complementing the changes in the XSD file).

File: src/strings-in-block/class-attributes.php [hunk 1]
```diff
@@ -86,11 +87,15 @@ private function getMatchingConfigKey( $attr_key, array $config_keys ) {
...
 	 *
 	 * @return string
 	 */
-	private function getRegex( $config_key ) {
-		if ( @preg_match( $config_key, '' ) !== false ) {
+	private function getRegex( $config_key, array $attrs ) {
+		if (
+			isset( $attrs['search-method'] )
+			&& \WPML_Gutenberg_Config_Option::SEARCH_METHOD_REGEX === $attrs['search-method']
+		) {
```
Comment (brucepearson) [inline src/strings-in-block/class-attributes.php]: @strategio How about:
```
if( $this->is_regex_attr( $attrs ) ) {
    return $config_key;
}
...
```

File: src/strings-in-block/class-attributes.php [hunk 2]
```diff
@@ -133,10 +138,11 @@ public function updateStringsRecursively( array $attrs, array $config_keys, arra
 			}
 
 			if ( is_array( $attr_value ) ) {
-				$next_level_keys = is_array( $config_keys[ $matching_key ] )
-					? $config_keys[ $matching_key ] : [ '*' => 1 ];
+				$children_config_keys = isset( $config_keys[ $matching_key ]['children'] )
+					? $config_keys[ $matching_key ]['children']
+					: $this->getMatchAllKey();
```
Comment (brucepearson) [inline src/strings-in-block/class-attributes.php]: @strategio How about:
```
$children_config_keys = $this->get_children_config_keys( $config_keys, $matching_key );
```

**Review by brucepearson — APPROVED** [review]
@strategio I've suggested some minor refactoring. It would be nice if the tests were clearer because it's hard to understand with all the `key`, `key12`, etc. +1

File: src/strings-in-block/class-attributes.php (hunk 2 above)
Comment (strategio) [reply src/strings-in-block/class-attributes.php]: Good idea @brucepearson, added!

File: src/strings-in-block/class-attributes.php (hunk 1 above)
Comment (strategio) [reply src/strings-in-block/class-attributes.php]: done

**Comment by strategio** [comment]
@brucepearson, I tried to make the tests as clearer as possible, but also I had to cover many different cases and this ends up with a lot of keys (I could not find a better way).

**Pull request merged**